[Bug] Cannot find module 'vuepress/package.json' when using version` 2.0.0-beta.184` or higher

### Checklist

- [X] I am using **LTS** version of Node.js.
- [X] I am using the **latest v2 version** of VuePress and VuePress official plugins.
- [X] I am using the **latest v2 version** of vuepress-theme-hope and all plugins in this repo.
- [X] I am following the docs and I am sure I am using the correct configuration.
- [X] I have searched [FAQ](https://theme-hope.vuejs.press/faq/)

### Package name

vuepress-theme-hope

### Which operating system are you using?

- [ ] macOS
- [X] Windows
- [ ] Linux

### Describe the bug

# 概述
今天更新到版本`2.0.0-beta.184`后出现了这个问题，测试发现最新的`beta.185`问题仍然存在，回退到`beta.183`后问题解决
具体报错如下：
``` console
PS E:\documents\repo\core-js> npm run docs:dev

> core-js@3.28.0 docs:dev
> npm run --workspace=docs dev


> dev
> vuepress dev

Error: Cannot find module 'vuepress/package.json'
Require stack:
- E:/documents/repo/core-js/docs
    at Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
    at Module._load (node:internal/modules/cjs/loader:885:27)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at file:///E:/documents/repo/core-js/docs/node_modules/vuepress-theme-hope/lib/node/index.js:1:6653
    at Array.forEach (<anonymous>)
    at ca (file:///E:/documents/repo/core-js/docs/node_modules/vuepress-theme-hope/lib/node/index.js:1:6624)
    at file:///E:/documents/repo/core-js/docs/node_modules/vuepress-theme-hope/lib/node/index.js:83:171
    at resolvePluginObject (file:///E:/documents/repo/core-js/node_modules/@vuepress/core/dist/index.js:633:65)
    at resolveThemeInfo (file:///E:/documents/repo/core-js/node_modules/@vuepress/core/dist/index.js:925:23)
npm ERR! Lifecycle script `dev` failed with error: 
npm ERR! Error: command failed
npm ERR!   in workspace: corejs-doc
npm ERR!   at location: E:\documents\repo\core-js\docs
```
# 问题追踪
根据关键词在src/node中查找，最后查找到这个commit: 
https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/3e8f1c80253839c1b87f7571c30b65efd85b0102 创建了`packages/theme/src/node/check/vuepress.ts`，问题代码在第48行:
``` ts
mainPackages.forEach((pkg) => {
    const { version } = <PackageJSON>require(`${pkg}/package.json`);

    mainPackagesVersions.push(version);
  });
```

### Additional context

_No response_

--- a/packages/shared/src/node/utils/version.ts
+++ b/packages/shared/src/node/utils/version.ts
@@ -16,7 +16,7 @@ interface PackageJSON extends Record<string, unknown> {
 export const checkVersion = (app: App, name: string, range = "v2"): void => {
   const sourceFolderPath = app.dir.source();
   const logger = new Logger(name);
-  const require = createRequire(sourceFolderPath);
+  const require = createRequire(`${sourceFolderPath}/`);
 
   let dir = sourceFolderPath;
   let packageName = "";

